### PR TITLE
Fixing ESLint in docs. (4/n, 89% done)

### PR DESCRIPTION
**Pull request by anthonyshew**
## Description
4/n of my stacked diffs to address ESLint errors in docs. PRs to ship before this one:
- https://github.com/vercel/turbo/pull/6812
- #6813
- #6818

### CODE REVIEW

File: docs/components/Tweet.tsx
Comment (anthonyshew) [inline docs/components/Tweet.tsx:1]: This file was not being used.

File: docs/components/HeaderLogo.tsx [hunk 1]
```diff
@@ -32,5 +32,3 @@ function HeaderLogo() {
     </>
   );
 }
-
-export default HeaderLogo;
```
Comment (mehulkar) [inline docs/components/HeaderLogo.tsx:36]: Very stylistic so I shouldn't care, but I like this change. I always miss it when there all the exports are at the bottom (some people really like this style and that's totally fine too)

File: docs/components/Navigation.tsx [hunk 2]
```diff
@@ -1,7 +1,8 @@
+/* eslint-disable -- Lots of Nextra magic. */
```
Comment (mehulkar) [inline docs/components/Navigation.tsx:1]: this feels like a prettttttty big hammer. is it worth it to list out all the rules instead?

**Review by mehulkar — APPROVED** [review]
👏🏾 👏🏾 👏🏾

**Review by tknickman — APPROVED** [review]

File: docs/components/HeaderLogo.tsx (hunk 1 above)
Comment (anthonyshew) [reply docs/components/HeaderLogo.tsx:36]: CleanShot 2023-12-20 at 11 25 28@2x

File: docs/components/Navigation.tsx (hunk 2 above)
Comment (anthonyshew) [reply docs/components/Navigation.tsx]: Oh, yup, good call.

**Pull request merged**